m5/src/repositories/commentRepository.ts

```diff
@@ -0,0 +1,45 @@
+import { Prisma } from '@prisma/client';
+import prisma from '../config/prisma.js';
+import type { CreateCommentData, UpdateCommentData } from '../types/comment.js';
+
+type PrismaTransaction = Prisma.TransactionClient;
+
+// 새 댓글 생성
+export const createComment = async (data: CreateCommentData, tx: PrismaTransaction = prisma) => {
+    return tx.comment.create({ data });
+};
+
+// 특정 리소스의 댓글 목록 조회
+export const findCommentsByResourceId = async (
+    resourceType: string,
+    resourceId: number,
+    tx: PrismaTransaction = prisma,
+) => {
+    const whereCondition = resourceType === 'post' ? { postId: resourceId } : { productId: resourceId };
```

Comment:
아래에서 resourceType으로 분기를 하고 싶다면, 유니온타입을 이용해서 다음과 같이 작성해보세요.
```suggestion
    resourceType: 'post' | 'product',
    resourceId: number,
    tx: PrismaTransaction = prisma,
) => {
    const whereCondition = resourceType === 'post' ? { postId: resourceId } : { productId: resourceId };
```